test/integration/filters/repick_cluster_filter.cc

```diff
@@ -0,0 +1,53 @@
+#include "test/integration/filters/repick_cluster_filter.h"
+
+#include <string>
+
+#include "envoy/http/filter.h"
+#include "envoy/registry/registry.h"
+#include "envoy/server/filter_config.h"
+
+#include "source/extensions/filters/http/common/pass_through_filter.h"
+
+#include "test/extensions/filters/http/common/empty_http_filter_config.h"
+
+#include "absl/strings/str_format.h"
+
+namespace Envoy {
+namespace RepickClusterFilter {
+
+// A test filter that modifies the request header (i.e. map the cluster header
+// to cluster name), clear the route cache, then will trigger the cluster
+// re-pick at router filter later in the filter chain.
```

Comment:
Seems like all this filter does is insert a single header (?) and then clears out the route cache, so I think this comment can be simplified?